app/frontend/wallet/shelley/shelley-transaction-planner.ts

```diff
@@ -149,24 +150,45 @@ function computeRequiredDeposit(certificates: Array<TxCertificate>): Lovelace {
   return certificates.reduce((acc, {type}) => acc + CertificateDeposit[type], 0) as Lovelace
 }
 
-export const computeMinUTxOLovelaceAmount = (
-  address: Address,
-  coins: Lovelace,
-  tokens: Token[]
-): Lovelace => {
+export const computeMinUTxOLovelaceAmount = (tokenBundle: TokenBundle): Lovelace => {
+  // based on https://github.com/input-output-hk/cardano-ledger-specs/blob/master/doc/explanations/min-utxo.rst
+  const quot = (x: number, y: number) => Math.floor(x / y)
+  const roundupBytesToWords = (x: number) => quot(x + 7, 8)
   // TODO: this to network config or constants
-  const adaOnlyUTxOSize = 74 // TODO: this is too much
   const minUTxOValue = 1000000
-  if (tokens.length === 0) return minUTxOValue as Lovelace
-  const output: TxOutput = {
-    isChange: false,
-    address,
-    coins,
-    tokens,
+  // NOTE: should be 2, but a bug in Haskell set this to 0
+  const coinSize = 0
+  const txOutLenNoVal = 14
+  const txInLen = 7
+  const utxoEntrySizeWithoutVal = 6 + txOutLenNoVal + txInLen // 27
+
+  // NOTE: should be 29 but a bug in Haskell set this to 27
+  const adaOnlyUtxoSize = utxoEntrySizeWithoutVal + coinSize
+
+  const distinctAssets = distinct(tokenBundle.map(({assetName}) => assetName))
+
+  const numAssets = distinctAssets.length
+  // number of unique policyIds
```

Comment:
this comment seems avoidable if we just rename the variable to numPolicyIds and from the assignment it's pretty obvious those are unique

Reply:
OK, but I still dont really understand why we want to change this implementation in anyway since it specifically points to the specification, and anyone wondering can just open the link and understand.

Reply:
It makes more sense to me to follow the spec to the letter rather than rename stuff just so it makes a bit more sense in the scope of the function, but it doesn't match whats in the spec.